### ISSUE TITLE
wrong gradient for mod?

### ISSUE THREAD

**Issue by etienne87** [philipperemy/tensorflow-phased-lstm#9]
In phased_lstm.py, gradient of modulo is:
```
@ops.RegisterGradient('FloorMod')
def _mod_grad(op, grad):
    x, y = op.inputs
    gz = grad
    x_grad = gz
    y_grad = tf.reduce_mean(-(x // y) * gz, axis=[0], keep_dims=True)[0]
    return x_grad, y_grad
```
Why reduce_mean and not reduce_sum? I understand that broadcasting y to x size makes multiple copies, so that gradient has to be summed over multiple usage, like in a convolution?

**Comment by philipperemy** [comment]
Yes I agree it should be `tf.reduce_sum`. Good catch :)

It's no longer needed as its now part of tensorflow.
